providers/gpgpu/tools/gpu-setup

```diff
@@ -65,6 +66,17 @@ apt install -y --no-install-recommends cuda-toolkit
 CUDA_PATH=$(find /usr/local -maxdepth 1 -type d -iname "cuda*")/bin
 export PATH=$PATH:$CUDA_PATH
 
+# clone cuda-samples repo
+echo "*  Cloning cuda-samples repo"
+CUDA_SAMPLES_DIR=/opt/cuda-samples
```

Comment:
My best understanding is that these are executables of some sort that run a test. If that is the case, the correct location to put them is under `providers/gpgpu/bin` after you build them. This makes it so you can call them directly "by name" from a Checkbox test (as the bin directory is automagically added to your PATH). If you do it this way, they will never work outside of a debian package/source run. If you need some data along with the binaries you should store it in `providers/gpgpu/data` (you can get the path of this directory from a test via the `PLAINBOX_PROVIDER_DATA` variable)

Also note that we support more than arm64 and amd64 (we build for a lot of other archs like armhf but also riscv64 or ppc, what does the script do under those circumstances?)

Reply:
The script is now building into `providers/gpgpu/bin`. I've added wrappers to address some limitations of nvidia's cuda-samples programs. See aca24be.

As to the architectures, the architecture used to be hardcoded to x86_64 before. They support arm64 in *some* cases, but not in others (eg Ubuntu 24.04); since it technically is supported in some cases, I added the option to use arm64, but I'm not sure if that's the best option. Let me know what you think (ie if we should limit this provider to x86_64 for now). See a854201.